library/std/src/io/stdio.rs

```diff
@@ -712,26 +705,37 @@ impl Write for &Stdout {
     }
 }
 
+impl StdoutLock<'_> {
+    #[inline]
+    fn with_inner<F, R>(&mut self, f: F) -> R
+    where
+        F: FnOnce(&mut LineWriter<StdoutRaw>) -> R,
+    {
+        let mut inner = self.inner.borrow_mut();
+        f(inner.get_or_insert_with(|| LineWriter::new(stdout_raw())))
```

Comment:
I think we should be able to make this a helper function rather than a callback with [RefMut::map](https://doc.rust-lang.org/nightly/std/cell/struct.RefMut.html#method.map), which will let you return the drop guard.

Reply:
Oh, I didn't see that. Thanks!